programs/vote/src/vote_state/mod.rs

```diff
@@ -226,6 +279,172 @@ impl VoteStateUpdate {
     }
 }
 
+/// Ignoring overhead, in a full `VoteStateUpdate` the lockouts take up
+/// 31 * (64 + 32) = 2976 bits.
+///
+/// In this schema we separate the votes into 3 separate lockout structures
+/// and store offsets rather than slot number, allowing us to use smaller fields.
+///
+/// In a full `CompactVoteStateUpdate` the lockouts take up
+/// (64 + 8) + (32 + 8) * 16 + (16 + 8) * 8 + (8 + 8) * 6 = 1000 bits
+/// allowing us to greatly reduce block size.
+#[frozen_abi(digest = "BfeSJNsfQeX6JU7dmezv1s1aSvR5SoyxKRRZ4ubTh2mt")]
+#[derive(Serialize, Default, Deserialize, Debug, PartialEq, Eq, Clone, AbiExample)]
+pub struct CompactVoteStateUpdate {
+    /// The proposed root
+    pub root: Option<Slot>,
```

Comment:
8 bytes for a boolean kinda hurts. Use u64::MAX to represent None?

Reply:
@carllin apart from tests, are there any valid cases where we need to be able to send None as the proposed root?

Reply:
when somebody has just started voting and hasn't established a root yet, but the default there as zero is probably fine